src/linear_algebra/matrix.lean

```diff
@@ -936,3 +936,53 @@ def alg_equiv_matrix {R : Type v} {M : Type w} {n : Type*} [fintype n]
   [comm_ring R] [add_comm_group M] [module R M] [decidable_eq n] {b : n → M} (h : is_basis R b) :
   module.End R M ≃ₐ[R] matrix n n R :=
 h.equiv_fun.alg_conj.trans alg_equiv_matrix'
+
+/-- `std_basis_vec` is the standard basis for the free module `n → R`. -/
+def std_basis_vec {R : Type v} [ring R] {n : Type w} [decidable_eq n]
+  (i : n) : n → R := λ j, if j = i then 1 else 0
```

Comment:
It's tempting to just define this directly in terms of `std_basis`,
```suggestion
/-- `std_basis_vec` is the standard basis for the free module `n → R`. -/
def std_basis_vec {R : Type v} [ring R] {n : Type w} [decidable_eq n]
  (i : n) : n → R := linear_map.std_basis R (λ _, R) i 1
```

Reply:
Or as `update`:

```suggestion
/-- `std_basis_vec` is the standard basis for the free module `n → R`. -/
def std_basis_vec {R : Type v} [ring R] {n : Type w} [decidable_eq n]
  (i : n) : n → R := function.update 0 i (1 : R)
```

Reply:
Should we define it as `linear_map.std_basis` and then prove it's a if else statement? I find working with the latter easier.

Reply:
Do you ever actually use the definition here? The lemmas you prove with if_pos and if_neg before are already proven for single and std_basis anyway